syntax = "proto3";

package proxy;

service ProxyService {
  rpc RegisterServer (ServerInfo) returns (ProxyResponse);
  rpc GetAvailableServers (Empty) returns (ServerList);
  rpc DeleteServer (ServerInfo) returns (ProxyResponse);
}

message ServerInfo {
  string ip = 1;
  int32 port = 2;
  string gpu_name = 3;
}

message ProxyResponse {
  string message = 1;
}

message Empty {}

message ServerList {
  repeated ServerInfo servers = 1;
}
